### ISSUE TITLE
Webpacker 4.0 can't resolve fonts

### ISSUE THREAD

**Issue by pedrojioc** [rails/webpacker#2155]
I am trying to add socicon to my project. I am using rails 6 and webpacker 4.

yarn add socicon
And I created a file vendors.scss
**#app/javascript/stylesheets/vendors/vendors.scss**
`@import "~socicon/css/socicon";`

**and I add file vendors.js to pack folder.**
`import '../stylesheets/vendors/vendors`'

**and I receiving this:**
ERROR in ./app/javascript/stylesheets/theme/vendors/vendors.scss (./node_modules/css-loader/dist/cjs.js??ref--8-1!./node_modules/postcss-loader/src??ref--8-2!./node_modules/sass-loader/lib/loader.js??ref--8-3!./app/javascript/stylesheets/vendors/vendors.scss)
Module not found: Error: Can't resolve '../font/socicon.eot' in '/home/p***/Documents/Projects/B***/app/javascript/stylesheets/vendors'

**Comment by jakeNiemiec** [comment]
Did you add the ‘- .eot’ extension to your ‘config/webpacker.yml’ file? I can’t remember if that’s one of the stock extensions. 

If that’s not the issue, could you post your package.json, environment.js and webpacker.yml files to troubleshoot?

**Comment by pedrojioc** [comment]
Yes, I added the extension '.eot'.

The same goes for the extensions '.svg, .ttf, .woff'

Files:

**# package.json**
```
{
  "name": "B***",
  "private": true,
  "dependencies": {
    "@fortawesome/fontawesome-free": "^5.9.0",
    "@fortawesome/free-solid-svg-icons": "^5.9.0",
    "@rails/actioncable": "^6.0.0-alpha",
    "@rails/activestorage": "^6.0.0-alpha",
    "@rails/ujs": "^6.0.0-alpha",
    "@rails/webpacker": "^4.0.7",
    "bootstrap": "^4.3.1",
    "jquery": "^3.4.1",
    "line-awesome": "^1.0.2",
    "perfect-scrollbar": "^1.4.0",
    "popper.js": "^1.15.0",
    "socicon": "^3.0.5",
    "turbolinks": "^5.2.0"
  },
  "version": "0.1.0",
  "devDependencies": {
    "url-loader": "^2.0.1",
    "webpack-dev-server": "^3.7.2"
  }
}
```
**# environment.js**
```
const { environment } = require('@rails/webpacker')
const webpack = require('webpack')

environment.plugins.append('Provide', new webpack.ProvidePlugin({
  $: 'jquery',
  jQuery: 'jquery',
  Popper: ['popper.js', 'default']
}))

// environment.plugins.append('Environment', new webpack.config.optimization.splitChunks({
//   name: 'vendor',
//   minChunks: Infinity,
// }))
// const urlLoader = {
//   test: /\.jpg|png|gif|woff|eot|ttf|svg|mp4|webm$/,
//   use: {
//     loader: 'url-loader',
//     options: {
//       limit: 90000,
//     }
//   }
// }
// environment.loaders.prepend('json', urlLoader)
// environment.loaders.get('sass').use.splice(-1, 0, {
//   loader: 'resolve-url-loader'
// });
module.exports = environment
```
**# webpacker.yml**

default: &default
  source_path: app/javascript
  source_entry_path: packs
  public_root_path: public
  public_output_path: packs
  cache_path: tmp/cache/webpacker
  check_yarn_integrity: false
  webpack_compile_output: false


  resolved_paths: []


  cache_manifest: false


  extract_css: false

  static_assets_extensions:
    - .jpg
    - .jpeg
    - .png
    - .gif
    - .tiff
    - .ico
    - .svg
    - .eot
    - .otf
    - .ttf
    - .woff
    - .woff2

  extensions:
    - .mjs
    - .js
    - .sass
    - .scss
    - .css
    - .module.sass
    - .module.scss
    - .module.css
    - .png
    - .svg
    - .gif
    - .jpeg
    - .jpg

development:
  <<: *default
  compile: true

  check_yarn_integrity: true


  dev_server:
    https: false
    host: localhost
    port: 3035
    public: localhost:3035
    hmr: false

    inline: true
    overlay: true
    compress: true
    disable_host_check: true
    use_local_ip: false
    quiet: false
    headers:
      'Access-Control-Allow-Origin': '*'
    watch_options:
      ignored: '**/node_modules/**'


test:
  <<: *default
  compile: true


  public_output_path: packs-test

production:
  <<: *default


  compile: false


  extract_css: true


  cache_manifest: true

**Comment by jakeNiemiec** [comment]
Move that ‘url-loader’ from devDependencies to dependencies and run ‘yarn install’. That has a chance of fixing things altogether.

**Comment by pedrojioc** [comment]
That did not work for me.

According to the webpacker documentation the sources will be compiled automatically, but this apparently is not working.

I added the url-loader to try to solve the problem but according to the documentation it should not do this as it should be ready

**Comment by pedrojioc** [comment]
I found a solution for this.
Apparently it is a problem with the relative routes. The css file is calling the ttf, eot, etc files within it and webpack was not able to resolve these routes.

The solution below.
[resolve-url-loader](https://github.com/rails/webpacker/blob/master/docs/css.md#resolve-url-loader)

And I proceed to close.
Thanks @jakeNiemiec

**Comment by Sibicle** [comment]
thanks for pointing me in the right direction! sadly that link no longer works, but I was able to piece together the solution.

in short, I changed my `config/webpack/environment.js` file to look like this:

```js
const { environment } = require('@rails/webpacker')

// added these three lines here, rest of the file was the default
environment.loaders.get('sass').use.splice(-1, 0, {
  loader: 'resolve-url-loader'
})

module.exports = environment
```

this fixed my issue!

possibly worth noting that if you upgrade to `webpacker` 6, you will need to update this file, but there are [instructions](https://github.com/rails/webpacker/blob/master/docs/v6_upgrade.md) related to the `environment` import. @gencer has a suggested solution to this [same problem](https://github.com/rails/webpacker/issues/2844#issuecomment-753489020). however, I didn't test the new `webpacker` yet.
